### DISCUSSION TITLE
Single animal tracking - how should I check my model worked before publication?

### DISCUSSION THREAD

**Discussion by storksle** [talmolab/sleap#2814] [Help!] [answered]
I am running a single animals tracking model through SLEAP. The model seems pretty good after repeated training and running it. I have the evaluation metrics below. What steps are traditionally taken to "proofread" such a model before getting read to submit for publication? It seems like most of the documentation I can find is related to proofreading with multiple animals in the videos and correcting identity issues related to that. Any help appreciate for a SLEAP newbie! Thanks!

<img width="356" height="206" alt="Screenshot 2026-07-06 120535" src="https://github.com/user-attachments/assets/617b13d3-3123-4722-bc46-28bcb9fa5cd2" />

Levi Storks

**Comment by tom21100227** [comment, marked as answer]
Hi Levi @storksle , 

Usually to show that your model is working well (aside from visually checking that it's producing good tracks), you would report metrics on test set and validation set (e.g. your models performance on data it was never trained/evaluate on). SLEAP does this split and validation for you automatically. Under your trained model's directory you will find something like `test_pred_metrics.npz` and/or `val_0_pred_metrics.npz`). 

Some useful metrics people tend to look at are distance error (_how far away (in pixels) is the model's prediction to it's actual location_) and mOKS (Object Keypoint Similarity). 

If you have multiple iterations of your model **and all of them have the same test/validation set**, a figure where the x axis is training data and y axis being those metrics would be nice (see below). **DO NOT COMPARE evaluation metrics calculated from different dataset**, if you're not sure if your models have the same validation set, you can label an additional 20-50 frames and use that slp file as your val set, `sleap-nn eval` is the tool to use. 

Some figure for inspo: 
<img width="1848" height="538" alt="image" src="https://github.com/user-attachments/assets/44ef1653-7076-44cd-9ff0-6f3992972d95" />

<img width="2574" height="964" alt="image" src="https://github.com/user-attachments/assets/032ed064-94be-4496-bf94-706297678b00" />

[(Pereira et al. 2019, Fig. 2d-2g)](https://www.nature.com/articles/s41592-022-01426-1/figures/2)



Useful links: 
[What all the evaluation metrics mean](https://nn.sleap.ai/latest/reference/evaluation_metrics)

<img width="1498" height="1162" alt="image" src="https://github.com/user-attachments/assets/21f58ddf-1728-4975-9ef5-f2d7056655f7" />

Hope this helps! Congrats on publishing! 

Best,
Tom

**Reply by storksle** [reply]
Thank you so much for your help!
